### PR TITLE
Deprecate subtype comparison operators

### PR DESCRIPTION
**Pull request by stonea**
This PR moves the subtype comparison operators to being deprecated (previously it was unstable).

As described here: https://github.com/chapel-lang/chapel/issues/19363#issuecomment-1307516579 there is some concern that this might cause a lot of deprecation warnings in user code so we may want to consider adding a flag to suppress this specific warning or coming up with some kind of tool that can be used to automatically update expressions that trigger it.

### CODE REVIEW

**Comment by DanilaFe** [comment]
> This PR moves the subtype comparison operators from being deprecated to being unstable.

Did you perhaps mean "from unstable to deprecated"?

**Comment by stonea** [comment]
> Did you perhaps mean "from unstable to deprecated"? 

Yes, whoops, I've updated the description.

**Comment by stonea** [comment]
We have a patch to update CHAMPS for this change here: https://github.com/chapelu-champs/champs-nightly/blob/main/patches/optional/deprecate_type_comparison.patch

I don't believe this impacts any other user codes.

File: test/deprecated/typeQueryFuncs.chpl [hunk 1]
```diff
@@ -39,3 +39,27 @@ writeln("isFloat(1.2) returns ", isFloat(1.2));
 writeln("isFloat(real) returns ", isFloat(real));
 writeln("isFloatType(real) returns ", isFloatType(real));
 writeln("isFloatValue(1.2) returns ", isFloatValue(1.2));
+
+// Type subtype relations (operators now marked unstable, will likely be
```
Comment (lydia-duncan) [inline test/deprecated/typeQueryFuncs.chpl:43]: Looks like this comment needs to be updated to say "deprecated" instead of "unstable"

**Review by lydia-duncan — APPROVED** [review]
One minor copy+paste error, otherwise this looks good to me!

**Pull request merged**